### PR TITLE
Ensure SEGIntegration initializes with Adobe

### PR DESCRIPTION
**Pull request by ladanazita**
This is what I believe was originally wrong with my implementation after end to end testing.

The original issue surfaced was after configuring Adobe with sample app in the repo, I set a break point on the [createWithProperties](https://github.com/segment-integrations/analytics-ios-integration-adobe-analytics/blob/master/Pod/Classes/SEGAdobeIntegration.m#L43) method, to ensure that this was being called. The tests were setup to validate against the ADBMediaObject mock, but the e2e tests failed to hit that break point, telling me my actual logic doesn't not reach that point.  

E2E Test setup:

```
#import <Segment-Adobe-Analytics/SEGAdobeIntegrationFactory.h>

@implementation SEGAdobeAppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
    SEGAnalyticsConfiguration *configuration = [SEGAnalyticsConfiguration configurationWithWriteKey:@"<my write key>"];
    configuration.trackApplicationLifecycleEvents = NO; // Enable this to record certain application events automatically!
    configuration.recordScreenViews = NO; // Enable this to record screen views automatically!
    [configuration use:[SEGAdobeIntegrationFactory instance]];
    [SEGAnalytics setupWithConfiguration:configuration];
    [SEGAnalytics debug:YES];

    [[SEGAnalytics sharedAnalytics] track:@"Video Playback Started" properties:@{
                                                                                 @"content_asset_id" : @"1234",
                                                                                 @"ad_type" : @"pre-roll",
                                                                                 @"video_player" : @"youtube",
                                                                                 } options: @{}];
    return YES;
}

```

I've implemented a solution that solves for not properly passing in the Adobe heartbeat instances (ADBObject, ADBMediaHeartbeat) when SEGAdobeIntegration factory was being created by passing in the mediheartbeat and mediaobject factories to SEGAdobeIntegrationFactory.m

Let me know if I'm way off here!

### CODE REVIEW

**Comment by ladanazita** [comment]
This works for now, but I need to refactor this
* Can have one factory AdobeObjectFactory build both ADBMediaHeartbeat and ADBMediaObject 
* Move into it's own file
* Have the podspec not include downloading this file to make it private

File: Pod/Classes/SEGAdobeIntegration.h [hunk 1]
```diff
@@ -40,7 +40,7 @@
 @property (nonatomic, strong, nullable) ADBMediaObject *mediaObject;
 @property (nonatomic, strong, nullable) id<SEGADBMediaObjectFactory> ADBMediaObjectFactory;
 
-- (instancetype _Nonnull)initWithSettings:(NSDictionary *_Nonnull)settings;
-- (instancetype _Nonnull)initWithSettings:(NSDictionary *_Nonnull)settings andADBMobile:(id _Nullable)ADBMobile andADBMediaHeartbeatFactory:(id<SEGADBMediaHeartbeatFactory> _Nullable)ADBMediaHeartbeatFactory andADBMediaHeartbeatConfig:(ADBMediaHeartbeatConfig *_Nullable)config andADBMediaObjectFactory:(id<SEGADBMediaObjectFactory> _Nullable)ADBMediaObjectFactory;
+//- (instancetype _Nonnull)initWithSettings:(NSDictionary *_Nonnull)settings;
+- (instancetype _Nonnull)initWithSettings:(NSDictionary *_Nonnull)settings andADBMobile:(id _Nullable)ADBMobileClass andADBMediaHeartbeat:(ADBMediaHeartbeat *_Nullable)ADBMediaHeartbeat andADBMediaHeartbeatFactory:(id<SEGADBMediaHeartbeatFactory> _Nullable)ADBMediaHeartbeatFactory andADBMediaHeartbeatConfig:(ADBMediaHeartbeatConfig *_Nullable)config andADBMediaObject:(ADBMediaObject *_Nullable)mediaObject andADBMediaObjectFactory:(id<SEGADBMediaObjectFactory> _Nullable)ADBMediaObjectFactory;
```
Comment (f2prateek) [inline Pod/Classes/SEGAdobeIntegration.h]: Why is this passing both `ADBMediaHeartbeat` and `ADBMediaHeartbeatFactory`? I understand you need one for tests and for the real implementation, but the factory should handle both cases.

**Comment by f2prateek** [comment]
> Can have one factory AdobeObjectFactory build both ADBMediaHeartbeat and ADBMediaObject
Not required, separate is fine.

> Move into it's own file
Also not required, same file is fine.

> Have the podspec not include downloading this file to make it private
What do you mean here? Which file? There are no podspec changes in this PR.

File: Pod/Classes/SEGAdobeIntegration.m [hunk 2]
```diff
@@ -460,6 +466,41 @@ - (void)trackHeartbeatEvents:(SEGTrackPayload *)payload
         SEGLog(@"[ADBMediaHeartbeat trackEvent:ADBMediaHeartbeatEventBufferStart mediaObject:%@ data:%@]", self.mediaObject, payload.properties);
         return;
     }
+
+    // Video Ad Break Started/Completed are not spec'd. For now, will document for Adobe and
+    // write a proposal to add this to the Video Spec
+    if ([payload.event isEqualToString:@"Video Ad Break Started"]) {
```
Comment (f2prateek) [inline Pod/Classes/SEGAdobeIntegration.m]: Think ad events should be added in their own PR. The PR is about initializing Adobe correctly - sending ad events seems like a different task.

File: Pod/Classes/SEGAdobeIntegration.m [hunk 3]
```diff
@@ -478,16 +519,18 @@ - (ADBMediaHeartbeatConfig *)createConfig:(SEGTrackPayload *)payload
...
         sslEnabled = true;
     }
 
-    bool debugEnabled = false;
+    BOOL debugEnabled = false;
     if (options[@"debug"]) {
         debugEnabled = true;
     }
 
+    //    self.config = [[ADBMediaHeartbeatConfig alloc] init];
```
Comment (f2prateek) [inline Pod/Classes/SEGAdobeIntegration.m]: Does leaving this commented like this serve a purpose?

File: Pod/Classes/SEGAdobeIntegration.m (hunk 2 above)
Comment (ladanazita) [reply Pod/Classes/SEGAdobeIntegration.m]: That change was already merged, I had this against the ad events branch to continue working, so Github just sees that as part of the change

File: Pod/Classes/SEGAdobeIntegration.m (hunk 3 above)
Comment (ladanazita) [reply Pod/Classes/SEGAdobeIntegration.m]: oh just missed this

**Comment by ladanazita** [comment]
@f2prateek  Made your requested changes, you were right the factory handled passing in the ADBMediaHeartbeat/Object, so I removed that. 

I also cleaned up the naming like you suggested

File: Example/Tests/Tests.m [hunk 4]
```diff
@@ -50,7 +50,7 @@ - (ADBMediaObject *)createWithProperties:(NSDictionary *)properties andEventType
 
     beforeEach(^{
         mockADBMobile = mockClass([ADBMobile class]);
-        integration = [[SEGAdobeIntegration alloc] initWithSettings:@{} andADBMobile:mockADBMobile andADBMediaHeartbeatFactory:nil andADBMediaHeartbeatConfig:nil andADBMediaObjectFactory:nil];
+        integration = [[SEGAdobeIntegration alloc] initWithSettings:@{} adobe:mockADBMobile andMediaHeartbeatFactory:nil andMediaHeartbeatConfig:nil andMediaObjectFactory:nil];
```
Comment (f2prateek) [inline Example/Tests/Tests.m]: Doesn't really look like this is used by most of the tests. It should be scoped better so we're only doing this for the tests where we need it.

Alternatively we can just pass in the mock factories here .
